WireDomain/Sources/WireDomain/Notifications/NotificationService.swift

```diff
@@ -130,60 +100,59 @@ final class NotificationService: UNNotificationServiceExtension {
         logger.info("finishing without showing notification")
         let emptyNotification = UNNotificationContent()
 
-        // With the "filtering" entitlement, we can tell iOS to not display a user notification by
-        // passing empty content to the content handler.
+        // With the "filtering" entitlement, we can tell iOS to not display a user notification by passing empty content
+        // to the content handler.
         // See https://developer.apple.com/documentation/bundleresources/entitlements/com_apple_developer_usernotifications_filtering
         contentHandler?(emptyNotification)
         terminate()
     }
 
+    private func terminate() {
+        // Content handler should only be consumed once.
+        contentHandler = nil
+        notificationSession = nil
+        onGoingTask = nil
```

Comment:
I have some doubts about this onGoingTask but this is out of scope of this PR.

question: As NSE can be reuse, do we really want to cancel the onGoingTask if a new notification arrives? just want to confirm this